pypi/pip_generate.go

```diff
@@ -567,18 +583,16 @@ func main() {
 
 	// Lastly, make `http_file` repo rules for PyPI links.
 	fmt.Fprintln(outputBzlFile, "def pypi_repositories():")
-	wroteAtLeastOne := false
+	// Don't call existing_rules repeatedly:
+	// https://github.com/bazelbuild/bazel/blob/master/site/docs/skylark/cookbook.md#aggregating-over-the-build-file
+	fmt.Fprintln(outputBzlFile, "    existing_rules = native.existing_rules()")
 	for _, dependency := range dependencies {
 		sort.Sort(wheelsByPlatform(dependency.wheels))
 		for _, wheel := range dependency.wheels {
 			if !wheel.useLocalWheel {
-				wroteAtLeastOne = true
 				name := wheel.bazelWorkspaceName(workspacePrefix)
 				fmt.Fprintf(outputBzlFile, wheel.makeBazelRule(&name, wheelDir))
 			}
 		}
 	}
-	if !wroteAtLeastOne {
```

Comment:
Why remove this chunk? If no targets have been generated, a `pass` is necessary, right?